crategen/models/tes_models.py

```diff
@@ -0,0 +1,283 @@
+"""Each model in this module conforms to the corresponding TES model names as specified by the GA4GH schema (https://ga4gh.github.io/task-execution-schemas/docs/)."""
+
+import os
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+from pydantic import AnyUrl, BaseModel, root_validator, validator
+
+from ..converters.utils import convert_to_iso8601
+
+
+class TESFileType(str, Enum):
+    """Enumeration of TES file types.
+
+    **Attributes:**
+
+    - **FILE** (`str`): Represents a file.
+    - **DIRECTORY** (`str`): Represents a directory.
+    """
+    FILE = "FILE"
+    DIRECTORY = "DIRECTORY"
+
+
+class TESState(str, Enum):
+    """Enumeration of TES task states.
+
+    **Attributes:**
+    - **UNKNOWN** (`str`): The task state is unknown.
+    - **QUEUED** (`str`): The task is queued.
+    - **INITIALIZING** (`str`): The task is initializing.
+    - **RUNNING** (`str`): The task is running.
+    - **PAUSED** (`str`): The task is paused.
+    - **COMPLETE** (`str`): The task is complete.
+    - **EXECUTOR_ERROR** (`str`): The task encountered an executor error.
+    - **SYSTEM_ERROR** (`str`): The task encountered a system error.
+    - **CANCELLED** (`str`): The task was cancelled.
```

Comment:
Please use Google-style docstrings, just like for the other code:
- No emphasis of section and attribute/arg names
- Indicate types in code (type hints), _not_ in docstrings

Reply:
I will revise the docstrings throughout `tes_models.py` to ensure they conform to the Google-style docstring guidelines used in our codebase. This includes:

- **Consistency:** Making sure the docstrings are consistent with other parts of the codebase.

- **Style:** Removing any emphasis (e.g., bold or italics) from section headers and attribute names.

- **Type Annotations:** Ensuring that types are specified in the code via type hints and not repeated in the docstrings.